### ISSUE TITLE
redundant_closure when using RangeInclusive syntax

### ISSUE THREAD

**Issue by conradludgate** [rust-lang/rust-clippy#10684]
### Summary

Clippy confuses `..=` for `RangeInclusive::new`.

### Lint Name

redundant_closure

### Reproducer

I tried this code:

```rust
fn test<T>(x: impl Fn(usize, usize) -> T) -> T { x(1, 2) } 
test(|start, end| start..=end)
```

I saw this happen:

```
warning: redundant closure
 --> src/main.rs:3:6
  |
3 | test(|start, end| start..=end)
  |      ^^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `start..=end`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure
  = note: `#[warn(clippy::redundant_closure)]` on by default
```

I expected to see this happen:

Either no warning, or suggesting `RangeInclusive::new` instead of range syntax

### Version

```text
rustc 1.69.0 (84c898d65 2023-04-16)
binary: rustc
commit-hash: 84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc
commit-date: 2023-04-16
host: aarch64-apple-darwin
release: 1.69.0
LLVM version: 15.0.7
```
